Update PGA doc for UAST dataset (#170)
src-d/datasets: PublicGitArchiveUASTs/ClickHouse/README.md

SELECT * 
FROM uasts
WHERE repo LIKE 'src-d/%'
    AND type = 'Comment';

# Extracting all identifiers from Go files, excluding vendoring files

SELECT * 
FROM uasts
WHERE lang = 'go'
    AND file NOT LIKE '%vendor/%'
    AND type = 'Identifier';
```

For more complex queries, e.g. extracting imports, some tricks are required (see [limitations](#limitations)) - however it can be [done](https://github.com/src-d/ml-mining).

### Origin

The DB was generated using [src-d/uast2clickhouse](https://github.com/src-d/uast2clickhouse) tool
from the original [Public GitArchive UASTs in Parquet](..).
The procedure took 3 days on 16 2-vcore, 16GB worker instances and a 64-vcore, 58GB server instance on
Google Cloud.

### Limitations

As was already mentioned in the [schema section](#schema), the UASTs are flattened in a lossy way.
This includes the aggressive normalization of the data for each programming language. While we did our
best at detecting the possible problems early on, a few sneaked inside when it was too late to fix them. Furthermore, the dataset was not updated after the 21st of september, so any commit merged after will not have been included.

> Below is a non-exhaustive list of known issues:

Reviewer: ```suggestion
Below is an incomplete list of known issues:
```

Author: done

* Duplication: 
    * Some repositories were processed multiple times, because they appeared in multiple Parquet files. We created the DB by iterating on each Parquet file, and did not check for duplicate rows.

>     * Errors inherited from Babelfish (language specific, e.g. [duplicate go comments](https://github.com/bblfsh/go-driver/issues/56)).

Reviewer: ```suggestion
    * Errors inherited from Babelfish that are language-specific, e.g. [duplicate go comments](https://github.com/bblfsh/go-driver/issues/56).
```

Author: done

* Missing data:

>     * Due to how we traversed the trees, in some cases data was purged, e.g. all Ruby imports. See [here](https://github.com/src-d/uast2clickhouse/issues/11).

Author: done

>     * Errors inherited from Babelfish (language specific, e.g. some drivers do not keep keywords).

Author: done

> * Wrong ordering: the left and right columns tend to be unreliable, due to how we traversed the UASTs and propagated positional information  - the line number is correct though.

Reviewer: ```suggestion
* Wrong ordering: the left and the right columns are sometimes unreliable, due to how we traversed the UASTs and propagated positional information. The line numbers should always be correct though.
```

Author: done

> If you see something strange or have trouble using the dataset, please say so [here](https://github.com/src-d/uast2clickhouse/issues) and we will look into it - however the dump will not be updated until Public Git Archive v3 is released.

Author: done
